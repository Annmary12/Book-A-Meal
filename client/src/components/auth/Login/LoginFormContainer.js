import { withFormik } from 'formik';
import { compose } from 'redux';
import { connect } from 'react-redux';
import * as Yup from 'yup';
import AuthForm from '../AuthForm.js';
import { login } from '../../../actions/auth.js';

const loginFormConfig = {
  mapPropsToValues: () => ({
    username: '',
    password: ''
  }),
  validationSchema: Yup.object().shape({
    username: Yup.string().required('username is required!'),
    password: Yup.string().required('Password is required')
  }),
  handleSubmit: (values, { setSubmitting, props }) => {
    const { from } = props.location.state || { from: '/menus' };

    props.dispatch(login(values, from));
    setSubmitting(false);
  }
};

const mapStateToProps = state => ({
  type: 'login',
  location: state.router.location
});

export default compose(connect(mapStateToProps), withFormik(loginFormConfig))(AuthForm);
