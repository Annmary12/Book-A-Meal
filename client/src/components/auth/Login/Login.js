import React from 'react';
import { Link } from 'react-router-dom';
import AuthArea from '../AuthArea.js';
import LoginFormContainer from './LoginFormContainer.js';


const Login = () => (
  <AuthArea>
    <LoginFormContainer />
    <p style={{ textAlign: 'center' }}>

      <Link to="/resetpassword" style={{ textAlign: 'center' }}>Forgot your password?</Link>
      <br /><br />
      <span style={{ textAlign: 'center' }}>Need an account? <Link to="/signup">Sign up</Link></span>
    </p>
  </AuthArea>
);

export default Login;
