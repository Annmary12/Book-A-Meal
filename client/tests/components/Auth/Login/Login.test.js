import React from 'react';
import Login from '../../../../src/components/Auth/Login/Login';

describe('Login component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Login />);

    expect(wrapper).toMatchSnapshot();
  })
})