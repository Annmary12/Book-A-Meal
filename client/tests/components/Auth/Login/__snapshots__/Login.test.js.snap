// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should match snapshot 1`] = `
<AuthArea>
  <Connect(C) />
  <p
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <Link
      replace={false}
      to="/reset_password"
    >
      Forgot your password?
    </Link>
    <br />
    <br />
    <span>
      Need an account? 
      <Link
        replace={false}
        to="/signup"
      >
        Sign up
      </Link>
    </span>
  </p>
</AuthArea>
`;
